### PR TITLE
Ddf solr master itestports memorydocs

### PR DESCRIPTION
**Pull request by rzwiefel**
### PR split into two commits for your reviewing pleasure
--------

#### What does this PR do?
 - Changes solr process ports in itests to hardcoded test port so it doesn't conflict with a running ddf process
- Adds documentation for solr memory settings (thanks to @mcalcote )

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: distribution/docs/src/main/resources/content/_running/starting-intro.adoc [hunk 1]
```diff
@@ -51,6 +51,15 @@ Comment out the -XX:+DisableAttachMechanism line and re-number remainder lines a
 ----
 ====
 
+.Solr Memory Considerations
+[NOTE]
+====
+From the https://lucene.apache.org/solr/guide/7_0/taking-solr-to-production.html[Solr Documentation]
+----
+For production, you’ll want to increase the maximum heap size based on the memory requirements of your search application; values between 10 and 20 gigabytes are not uncommon for production servers.
```
Comment (vinamartin) [inline distribution/docs/src/main/resources/content/_running/starting-intro.adoc]: This wording sounds a bit too casual considering the style of our documentation. 
`For production, increase the maximum heap size based on....`

Comment (rzwiefel) [reply distribution/docs/src/main/resources/content/_running/starting-intro.adoc]: This is straight excerpt from solr docs

Comment (rzwiefel) [reply distribution/docs/src/main/resources/content/_running/starting-intro.adoc]: 👼

Comment (rzwiefel) [reply distribution/docs/src/main/resources/content/_running/starting-intro.adoc]: Go back to army :trollface:

Comment (mcalcote) [reply distribution/docs/src/main/resources/content/_running/starting-intro.adoc]: The quote will also be in a box when rendered emphasising that the information comes from the Solr Docs linked above it.

File: distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/AbstractIntegrationTest.java [hunk 2]
```diff
@@ -693,7 +693,7 @@ private static Integer findPortNumber(Integer portToTry) {
     return options(
         editConfigurationFilePut(SYSTEM_PROPERTIES_REL_PATH, "solr.client", "HttpSolrClient"),
         editConfigurationFilePut(
-            SYSTEM_PROPERTIES_REL_PATH, "solr.http.url", "http://localhost:8994/solr"),
+            SYSTEM_PROPERTIES_REL_PATH, "solr.http.url", "http://localhost:9994/solr"),
```
Comment (mcalcote) [inline distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/AbstractIntegrationTest.java:696]: don't forget to add a line in README.md with this info

```
## Accessing Solr
Solr runs on port 9994 during the itests
```

Comment (rzwiefel) [reply distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/AbstractIntegrationTest.java:696]: Separate PR

Comment (rzwiefel) [reply distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/AbstractIntegrationTest.java:696]: https://github.com/rzwiefel/ddf-solr/pull/7

File: distribution/test/itests/test-itests-ddf/pom.xml [hunk 3]
```diff
@@ -379,13 +358,13 @@
...
-                                <argument>restart</argument>
+                                <argument>start</argument>
                                 <argument>-p</argument>
                                 <!--The maven properties plugin reads the system properties file-->
                                 <!--and provides the solr.http.port property from it.-->
-                                <argument>${solr.http.port}</argument>
+                                <argument>9994</argument>
                                 <argument>-m</argument>
-                                <argument>2g</argument>
+                                <argument>512m</argument>
```
Comment (mcalcote) [inline distribution/test/itests/test-itests-ddf/pom.xml:367]: 🤞

**Review by mcalcote — APPROVED** [review]

File: distribution/docs/src/main/resources/content/_running/starting-intro.adoc (hunk 1 above)
Comment (ahoffer) [reply distribution/docs/src/main/resources/content/_running/starting-intro.adoc]: 10-20 GB heap? I would think that would bring most JVM GCs to their knees.

Comment (ahoffer) [reply distribution/docs/src/main/resources/content/_running/starting-intro.adoc]: I feel this information does not belong in a quick start. It does not tell the user to take action, which something in a quick start should do. I'm reverting it.

**Review by ahoffer — APPROVED** [review]

**Review by ahoffer — APPROVED** [review]

**Pull request merged**